File: GO/FibonacciRapido.go
/**
 * @version GO
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */
package GO

import "math"

/**
 * Fibonacci value using Binet formule.
 */
func binetFormule(N int) float64 {
	return (math.Pow((1+math.Sqrt(5))/2, float64(N)) - math.Pow((1-math.Sqrt(5))/2, float64(N))) / math.Sqrt(5)
}

File: JS/FibonacciRapido.js
/**
 * @version JAVASCRIPT
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */

//console.log("Result is "+binetFormule(19));

/**
 * Fibonacci value using Binet formule.
 */
function binetFormule(N) {
    return +((Math.pow((1 + Math.sqrt(5)) / 2, N) - Math.pow((1 - Math.sqrt(5)) / 2, N)) / Math.sqrt(5)).toFixed(1);
}
module.exports = {binetFormule}
